lec13-projects-r.Rmd

```diff
@@ -0,0 +1,236 @@
+specially powerful when your project involves more than just a single file or
+has any type of code or data involved.
+
+[Git](https://git-scm.com/book/en/v2) is a software that keeps detailed tracking
+of your files and folders, keeping a history of exact characters that have been
+changed. It's very powerful, open source, and is very popular among software
+developers, programmers, R coders, and scientists. This is one of the main
+reasons why we will be learning Git, because it has this massive online
+community, support, and documentation. Git is especially powerful with
+collaborating on more complex projects (such as a research project).
+
+Git works by using several commands to track files ('add'), save ('commit') them
+to history, check the 'status' of your folders and files (in a project/parent
+folder, called a 'repo'), and see the history (the 'log') of what was saved. You
+can do some much more advanced and powerful things in Git, but this lesson we'll 
+focus on the basics (which you many only ever use).
+
+#### Exercise
+
+Follow the getting setup steps from command line:
+
+```bash
+git config --global user.name "Your Name"
+git config --global user.email yourname@domain.com
+```
+
+Replace "Your Name", with, well, your first and last name. Replace the fake
+email address with the email address you normally use.
+
+Confirm it works by running:
+
+```bash
+git config --list
+```
+
+### The basic commands of Git
+
+There are a few commands that you will use regularly when you start using Git.
+In fact, you may never need to use more than this depending on what you do.
+
+- `git add`
+- `git commit`
+- `git log`
+- `git status`
+
+For your final assignment, you will be using a few more commands that we will
+cover a bit in this lesson and much more next week.
+
+- `git remote`
+- `git push`
+- `git pull`
+- `git checkout`
+- `git branch` (maybe)
+
+But first, let's focus on the first four basic commands.
+
+- `git add` tells Git to start 'tracking' or watching the file(s) you point to
+and put them into the 'staging area' (see the image below).
+- `git commit` tells Git to take the changes made to the files you select and
+save them into the history, moving files from the 'staging area' into the
+history.
+- `git log` shows you the history of your git repository (all the changes made
+to the files; this is all contained in the `.git/` folder).
+- `git status` gives you a brief overview of what is going on in your git
+repository.
+
+![Image of the Stages from the [Git Basics Book](https://git-scm.com/book/en/v2/)](https://git-scm.com/book/en/v2/images/areas.png)
+
+RStudio has an excellent Git interface, mainly for the basic things. However,
+for slightly more complicated things you will need to use the terminal.
+So we'll start by learning how to use
+the terminal, then switch to using the RStudio interface.
+
+#### Exercise
+
+Create a new Markdown file called `bio.md` in the Git repo we created in class. Add your
```

Comment:
The repo from UofTCoders? Or locally (which hasn't been done in this lecture yet)?